src/containers/filedataset.jl

```diff
@@ -0,0 +1,57 @@
+matches(re::Regex) = f -> matches(re, f)
+matches(re::Regex, f) = !isnothing(match(re, f))
+const RE_IMAGEFILE = r".*\.(gif|jpe?g|tiff?|png|webp|bmp)$"i
+isimagefile(f) = matches(RE_IMAGEFILE, f)
+
+"""
+    rglob(filepattern, dir = pwd(), depth = 4)
+
+Recursive glob up to `depth` layers deep within `dir`.
+"""
+function rglob(filepattern = "*", dir = pwd(), depth = 4)
+    patterns = [repeat("*/", i) * filepattern for i in 0:(depth - 1)]
+
+    return vcat([glob(pattern, dir) for pattern in patterns]...)
+end
+
+"""
+    loadfile(file)
+
+Load a file from disk into the appropriate format.
+"""
+function loadfile(file::String)
```

Comment:
Not sure if a bunch of nested `if`s is the most elegant solution for this. Since this was meant to be a high-level function for FastAI.jl, maybe we can do without it for now?

Reply:
Should I just go with `FileIO.load` then?

Reply:
Yeah, I think that's the best for now. But let's re-export `load` instead of creating a new `loadfile` function